### PR TITLE
US135595 [AQD] Lit 2 migration - Phase 1

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US135595](https://rally1.rallydev.com/#/?detail=/userstory/623256112293&fdp=true): [AQD] Lit 2 migration - Phase 1

This PR makes backwards-compatible changes to prepare all repos for a Lit 2 upgrade. 

**Non-breaking change checklist:** 

- [x] Add missing `.js` extensions for lit-html or lit-element imports
- [x] `await requestUpdate` -> `await elem.updateComplete`
- [x] add non-null checks to shadowRoot accesses (not including test files)
- [x] ensure `undefined` is not passed into unsafeHTML calls
- [x] remove dynamic code from inside  `<template>` tags (Lit files only)
- [x] remove any dependencies on old Lit versions
  - [x] double-check by vetting `package-lock` file
- [x] audit `ifDefined` usages for null parameters

FYI @Brightspace/draco-vis

### CODE REVIEW

File: components/chart/chart.js [hunk 1]
```diff
@@ -182,7 +182,7 @@ class Chart extends SkeletonMixin(Localizer(LitElement)) {
 	}
 
 	get chartContainer() {
-		return this.shadowRoot.querySelector('#chart-container');
+		return this.shadowRoot?.querySelector('#chart-container');
```
[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply components/chart/chart.js:185]: Good point! Thank you. Checking on `null` and `undefined` downstream is doable. 

It seems that I had a false impression that our goal was just to shift the error point so that it was identifiable and doesn't look like access to undefined shadowRoot when we start using server-side rendering (SSR). 

@annabelsegalld2l, just a thought

It just occurs to me that adding `null`-checks is not ideal. What if we can add a test that makes SSR and fails when `shadowRoot` is null? Then that test can be used as part of efforts on migration on SSR.

[the remark this answers is not in the document]
Comment (annabelsegalld2l) [reply components/chart/chart.js:185]: The shadowRoot will always be null before the connectedCallback. The SSR capabilities were the reasoning motivation behind the change, so unfortunately when we flip to Lit 2 we will get errors for all the shadowRoot accesses before connectedCallback 😢 .

You are right that our goal is to shift the error point. We would also like to prevent any errors from occurring if possible. Here is an example PR where we stopped the error from occurring altogether by moving the shadowRoot access: https://github.com/Brightspace/outcomes-level-of-achievement-ui/pull/188/files.  I agree that adding null checks is a bit of a tedious solution. What I would recommend in your case is to audit those usages of the getter instead. If you can see that the caller will use the getter once the shadowRoot must be defined, you can skip the check there.  For example, if the caller is within the firstUpdated or updated methods, or the caller is an event listener. In those cases, the shadowRoot has to be defined. 

Unfortunately we haven't thought of any tests to check for this other than auditing the code. It's hard to tell when the code will be called, so we encouraged everyone to add "just-in-case" checks if they aren't super familiar with the code. 

So sorry for the wall of text! Hopefully that explains a bit of the motivation 😄

[the remark this answers is not in the document]
Comment (Vitalii-Misechko) [reply components/chart/chart.js:185]: Thank you for the detailed explanation. I prefer `a wall of text` instead of guessing 😄

**Review by johngwilkinson — APPROVED** [review]

**Review by annabelsegalld2l — APPROVED** [review]
LGTM!
